### PR TITLE
Implement `zero_vec` for packed fields

### PR DESCRIPTION
**Pull request by dlubarov**

### CODE REVIEW

File: goldilocks/src/lib.rs [hunk 1]
```diff
@@ -199,6 +199,12 @@ impl AbstractField for Goldilocks {
     fn generator() -> Self {
         Self::new(7)
     }
+
+    #[inline]
+    fn zero_vec(len: usize) -> Vec<Self> {
+        // SAFETY: repr(transparent) ensures transmutation safety.
+        unsafe { transmute(vec![0u64; len]) }
+    }
```
Comment (SyxtonPrime) [inline goldilocks/src/lib.rs:207]: Given how easy the change is, it's probably worth implementing it for the AVX2/AVX512 Goldilocks implementations as well.

File: monty-31/src/x86_64_avx2/packing.rs [hunk 2]
```diff
@@ -4,6 +4,7 @@ use core::mem::transmute;
 use core::ops::{Add, AddAssign, Div, Mul, MulAssign, Neg, Sub, SubAssign};
 
 use p3_field::{AbstractField, Field, PackedField, PackedFieldPow2, PackedValue};
+use p3_util::convert_vec;
 use rand::distributions::{Distribution, Standard};
 use rand::Rng;
```
Comment (SyxtonPrime) [inline monty-31/src/x86_64_avx2/packing.rs:9]: I think you also need to add: use alloc::vec::Vec;

File: monty-31/src/x86_64_avx512/packing.rs [hunk 3]
```diff
@@ -4,6 +4,7 @@ use core::mem::transmute;
 use core::ops::{Add, AddAssign, Div, Mul, MulAssign, Neg, Sub, SubAssign};
 
 use p3_field::{AbstractField, Field, PackedField, PackedFieldPow2, PackedValue};
+use p3_util::convert_vec;
 use rand::distributions::{Distribution, Standard};
 use rand::Rng;
```
Comment (SyxtonPrime) [inline monty-31/src/x86_64_avx512/packing.rs:9]: I think you also need to add: I think you also need to add: use alloc::vec::Vec;

**Review by SyxtonPrime — APPROVED** [review]
LGTM

File: util/src/lib.rs [hunk 4]
```diff
@@ -221,6 +222,29 @@ where
...
+/// # Safety
+/// In addition to the usual safety considerations around transmutation, the caller must ensure that
+/// the two types have the same alignment, that one of their sizes is a multiple of the other.
+#[inline(always)]
+pub unsafe fn convert_vec<T, U>(mut vec: Vec<T>) -> Vec<U> {
+    let ptr = vec.as_mut_ptr() as *mut U;
+    let len_bytes = vec.len() * size_of::<T>();
+    let cap_bytes = vec.capacity() * size_of::<T>();
+
+    debug_assert_eq!(align_of::<T>(), align_of::<U>());
```
[the remark this answers is not in the document]
Comment (dlubarov) [reply util/src/lib.rs:238]: Good point, done

File: util/src/lib.rs [hunk 5]
```diff
@@ -221,6 +222,29 @@ where
...
+    debug_assert_eq!(len_bytes % size_of::<U>(), 0);
+    debug_assert_eq!(cap_bytes % size_of::<U>(), 0);
```
[the remark this answers is not in the document]
Comment (dlubarov) [reply util/src/lib.rs:240]: Sounds reasonable, changed to `assert_eq`

File: monty-31/src/x86_64_avx2/packing.rs (hunk 2 above)
Comment (dlubarov) [reply monty-31/src/x86_64_avx2/packing.rs:10]: Good catch, fixed

File: monty-31/src/x86_64_avx512/packing.rs (hunk 3 above)
Comment (dlubarov) [reply monty-31/src/x86_64_avx512/packing.rs:10]: Good catch, fixed

**Pull request merged**